### PR TITLE
Improve setup of temporary local repository

### LINKED ISSUE #13: [feature] Use global local repository
@roberth commented on https://github.com/fzakaria/mvn2nix/pull/9#discussion_r475901251 the idea of using `~/.m2` as a remote repository to speed up the tool.

Furthermore, he suggested calculating the sha256 of the local file.
Both are great suggestions and will speed up the runtime of the tool tremendously.

### PR DESCRIPTION
**Pull request by fzakaria**
This comes as feedback by @roberth  that the temporary local repository should use the global local repository as an additional remote repository.

The SHA256 is also calculated from the local file rather than downloading it from the remote repository.
Overall much less network bandwidth for subsequent builds.

Here is an example of the output of _maven-invoker_ that demonstrates it downloading it from my local repository.
```bash
2020-08-24 14:54:48,371 [Thread-1] INFO : [INFO] Downloading from local.central: file:///home/fmzakari/.m2/repository/info/picocli/picocli-codegen/4.5.0/picocli-codegen-4.5.0.jar
2020-08-24 14:54:48,373 [Thread-1] INFO : [INFO] Downloaded from local.central: file:///home/fmzakari/.m2/repository/info/picocli/picocli-codegen/4.5.0/picocli-codegen-4.5.0.jar (128 kB at 43 MB/s)
2020-08-24 14:54:48,377 [Thread-1] INFO : [INFO] Downloading from local.central: file:///home/fmzakari/.m2/repository/info/picocli/picocli-codegen/4.5.0/picocli-codegen-4.5.0.pom
2020-08-24 14:54:48,379 [Thread-1] INFO : [INFO] Downloaded from local.central: file:///home/fmzakari/.m2/repository/info/picocli/picocli-codegen/4.5.0/picocli-codegen-4.5.0.pom (1.9 kB at 1.9 MB/s)
```

Fixes #13

### CODE REVIEW

**Comment by fzakaria** [comment]
@DerGuteMoritz @roberth tag.

**Comment by DerGuteMoritz** [comment]
Just gave it a try, works like a charm :ok_hand: Runtime is down from 63s to 14s on my test project :rocket:

**Comment by fzakaria** [comment]
> Just gave it a try, works like a charm  Runtime is down from 63s to 14s on my test project

Sweet -- I might commit then.

File: README.md [hunk 1]
```diff
@@ -94,14 +97,24 @@ in mkDerivation rec {
...
+
 ## Development
 
 If you are running *mvn2nix* from this repository, you can do so with **nix-build**
 
 ```bash
 $ nix-build
 
-./result/bin/mvn2nix > dependencies.nix
+./result/bin/mvn2nix > example/dependencies.nix
```
Comment (fzakaria) [inline README.md]:
```suggestion
./result/bin/mvn2nix > ./example/mvn2nix/dependencies.nix
```

**Pull request merged**